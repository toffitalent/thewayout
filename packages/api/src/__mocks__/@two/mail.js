const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
