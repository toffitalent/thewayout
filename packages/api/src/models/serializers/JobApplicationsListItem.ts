import { Serializer } from '@disruptive-labs/objection-plugins';
import {
  getCloakedData,
  getCloakedEmail,
  getFullyCloakedData,
  JobApplicationListItem,
  JobApplicationStatus,
} from '@two/shared';
import { getImageProxyUrl } from './utils';

export class JobApplicationListItemSerializer extends Serializer {
  static attributes = [
    'id',
    'jobId',
    'clientId',
    'status',
    'firstName',
    'lastName',
    'phone',
    'email',
    'avatar',
  ];

  firstName(value: JobApplicationListItem['firstName'], client: JobApplicationListItem) {
    return client.status === JobApplicationStatus.interview ||
      client.status === JobApplicationStatus.hired
      ? value
      : getCloakedData(value);
  }

  lastName(value: JobApplicationListItem['lastName'], client: JobApplicationListItem) {
    return client.status === JobApplicationStatus.interview ||
      client.status === JobApplicationStatus.hired
      ? value
      : getCloakedData(value);
  }

  phone(value: JobApplicationListItem['phone'], client: JobApplicationListItem) {
    if (value) {
      return client.status === JobApplicationStatus.interview ||
        client.status === JobApplicationStatus.hired
        ? value
        : getFullyCloakedData();
    }
    return undefined;
  }

  email(value: JobApplicationListItem['email'], client: JobApplicationListItem) {
    return client.status === JobApplicationStatus.interview ||
      client.status === JobApplicationStatus.hired
      ? value
      : getCloakedEmail(value);
  }

  avatar(value: JobApplicationListItem['avatar'], client: JobApplicationListItem) {
    return (client.status === JobApplicationStatus.interview ||
      client.status === JobApplicationStatus.hired) &&
      value
      ? getImageProxyUrl(value)
      : undefined;
  }
}
