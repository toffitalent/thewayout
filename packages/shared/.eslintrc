{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
