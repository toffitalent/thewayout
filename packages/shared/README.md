# @two/shared

Copyright (c) 2023-Present The Way Out, Inc. All Rights Reserved.

## Development

This package contains code shared between the [api](../api/README.md), [app](../app/README.md), and [web](../web/README.md) packages and is not intended to run in a standalone manner. See the READMEs of those packages for information on starting a development environment.

## Testing

To run all component and unit tests and get coverage reports, simply run `yarn workspace @dlatwobs/shared test`.
