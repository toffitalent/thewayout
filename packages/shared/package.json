{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
