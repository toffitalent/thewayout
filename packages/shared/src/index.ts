// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
