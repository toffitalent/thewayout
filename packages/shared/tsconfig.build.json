{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "incremental": true,
    "module": "commonjs",
    "noEmit": false,
    "outDir": "./build",
    "rootDir": "./src",
    "tsBuildInfoFile": "./build/.tsbuildinfo"
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["build", "node_modules", "**/*.test.ts", "**/*.test.tsx", "test"]
}
