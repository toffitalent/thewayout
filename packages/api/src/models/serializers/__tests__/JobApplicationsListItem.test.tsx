import { JobApplicationStatus } from '@two/shared';
import { JobApplicationListItemSerializer } from '../JobApplicationsListItem';

const mock = {
  id: 'TEST_ID',
  jobId: 'JOB_ID',
  clientId: 'CLIENT_ID',
  firstName: 'Firstname',
  lastName: 'Lastname',
  phone: '3345556666',
  email: 'test@email.com',
};

const cloakedStatuses = [
  JobApplicationStatus.applied,
  JobApplicationStatus.rejected,
  JobApplicationStatus.expired,
  JobApplicationStatus.notAFit,
];

const uncloakedStatuses = [JobApplicationStatus.hired, JobApplicationStatus.interview];

describe('JobApplicationListItemSerializer', () => {
  cloakedStatuses.forEach((status) => {
    test(`displays cloaked data for client with ${status} status`, async () => {
      const serializer = new JobApplicationListItemSerializer();

      const serialized = await serializer.serialize({ ...mock, status } as any);

      expect(serialized).toMatchSnapshot({ id: expect.any(String) });
    });
  });

  uncloakedStatuses.forEach((status) => {
    test(`displays uncloaked data for client with ${status} status`, async () => {
      const serializer = new JobApplicationListItemSerializer();

      const serialized = await serializer.serialize({ ...mock, status } as any);

      expect(serialized).toMatchSnapshot({ id: expect.any(String) });
    });
  });
});
