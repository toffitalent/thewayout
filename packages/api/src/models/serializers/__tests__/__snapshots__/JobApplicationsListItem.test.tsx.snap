// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobApplicationListItemSerializer displays cloaked data for client with applied status 1`] = `
{
  "clientId": "CLIENT_ID",
  "email": "t****@*****.***",
  "firstName": "F****",
  "id": Any<String>,
  "jobId": "JOB_ID",
  "lastName": "L****",
  "phone": "*****",
  "status": "applied",
}
`;

exports[`JobApplicationListItemSerializer displays cloaked data for client with expired status 1`] = `
{
  "clientId": "CLIENT_ID",
  "email": "t****@*****.***",
  "firstName": "F****",
  "id": Any<String>,
  "jobId": "JOB_ID",
  "lastName": "L****",
  "phone": "*****",
  "status": "expired",
}
`;

exports[`JobApplicationListItemSerializer displays cloaked data for client with notAFit status 1`] = `
{
  "clientId": "CLIENT_ID",
  "email": "t****@*****.***",
  "firstName": "F****",
  "id": Any<String>,
  "jobId": "JOB_ID",
  "lastName": "L****",
  "phone": "*****",
  "status": "notAFit",
}
`;

exports[`JobApplicationListItemSerializer displays cloaked data for client with rejected status 1`] = `
{
  "clientId": "CLIENT_ID",
  "email": "t****@*****.***",
  "firstName": "F****",
  "id": Any<String>,
  "jobId": "JOB_ID",
  "lastName": "L****",
  "phone": "*****",
  "status": "rejected",
}
`;

exports[`JobApplicationListItemSerializer displays uncloaked data for client with hired status 1`] = `
{
  "clientId": "CLIENT_ID",
  "email": "test@email.com",
  "firstName": "Firstname",
  "id": Any<String>,
  "jobId": "JOB_ID",
  "lastName": "Lastname",
  "phone": "3345556666",
  "status": "hired",
}
`;

exports[`JobApplicationListItemSerializer displays uncloaked data for client with interview status 1`] = `
{
  "clientId": "CLIENT_ID",
  "email": "test@email.com",
  "firstName": "Firstname",
  "id": Any<String>,
  "jobId": "JOB_ID",
  "lastName": "Lastname",
  "phone": "3345556666",
  "status": "interview",
}
`;
